### ISSUE TITLE
colloquial dialects

### ISSUE THREAD

**Issue by maherr13** [abjadai/catt#2]
Hello Thanks for your work, 

you mentioned in the paper that pretraining phase would help in performing good on colloquial dialects.

I wanted to ask how the model would respond to colloquial dialects something like finetuning the pretrained models on Egyptian dialects. 

if well, how much data do i need to finetune the model on to get fine good results on such a case.

**Comment by ahmedbr** [comment]
Hi, I have a similar issues actually. 

I've just used the encoder-decoder pretrained model to train a diacritizer for a gulf dialect. I didn't make major changes in the script provided, only provided datasets and pretrained model's paths. But as training proceeds, both val_loss and val_der were getting worse and worse. Please have a look at the screenshot below:

![image](https://github.com/user-attachments/assets/af2fb628-86da-4950-ae33-0e96288a02d4)

Did I do something wrong? Any explanation?

**Comment by farisalasmary** [comment]
Hi @maherr13 ,
The objective of the pretraining phase was to help the model in understanding the structural differences between classical Arabic and Saudi dialects. Since most of the labelled diacritization data is taken from books written in classical Arabic, the model wouldn't see those differences unless it is exposed to some dialect data. 
That was our intuition.

Regarding the data size for new dialects, I think it is hard to tell unless you try. Most Saudi dialects don't diverge too far from the classical Arabic words which might helped the model as well.

**Comment by farisalasmary** [comment]
Hi @ahmedbr,

Try freezing the weights of the encoder part as well as setting the learning rate to a lower value (e.g. 1e-5).

Also, the dataset might be a main contributor to the error if it has a completely different distribution compared to the original dataset that the model was trained on.
